game-core/src/main/java/games/strategy/engine/data/Route.java

```diff
@@ -404,25 +407,57 @@ public static Change getFuelChanges(final Collection<Unit> units, final Route ro
       final GameData data) {
     final CompositeChange changes = new CompositeChange();
     final Tuple<ResourceCollection, Set<Unit>> tuple =
-        Route.getFuelCostsAndUnitsChargedFlatFuelCost(units, route, player, data);
-    changes.add(ChangeFactory.removeResourceCollection(player, tuple.getFirst()));
-    for (final Unit unit : tuple.getSecond()) {
-      changes.add(ChangeFactory.unitPropertyChange(unit, Boolean.TRUE, TripleAUnit.CHARGED_FLAT_FUEL_COST));
+        Route.getFuelCostsAndUnitsChargedFlatFuelCost(units, route, player, data, false);
+    if (!tuple.getFirst().isEmpty()) {
+      changes.add(ChangeFactory.removeResourceCollection(player, tuple.getFirst()));
+      for (final Unit unit : tuple.getSecond()) {
+        changes.add(ChangeFactory.unitPropertyChange(unit, Boolean.TRUE, TripleAUnit.CHARGED_FLAT_FUEL_COST));
+      }
     }
     return changes;
   }
 
   public static ResourceCollection getMovementFuelCostCharge(final Collection<Unit> units, final Route route,
       final PlayerID player, final GameData data) {
-    return Route.getFuelCostsAndUnitsChargedFlatFuelCost(units, route, player, data).getFirst();
+    return Route.getFuelCostsAndUnitsChargedFlatFuelCost(units, route, player, data, false).getFirst();
+  }
+
+  /**
+   * Calculates how much fuel each player needs to scramble the specified units. ONLY SUPPORTS 1 territory distance
+   * scrambles properly as otherwise a route needs to be calculated.
+   */
+  public static Map<PlayerID, ResourceCollection> getScrambleFuelCostCharge(final Collection<Unit> units,
+      final Territory from, final Territory to, final GameData data) {
+    final Map<PlayerID, ResourceCollection> map = new HashMap<>();
+    final Route toRoute = new Route(from, to);
+    final Route returnRoute = new Route(to, from);
+    for (final Unit unit : units) {
+      final PlayerID player = unit.getOwner();
+      final ResourceCollection cost = new ResourceCollection(data);
+      cost.add(getMovementFuelCostCharge(Collections.singleton(unit), toRoute, player, data));
+      cost.add(getFuelCostsAndUnitsChargedFlatFuelCost(
+          Collections.singleton(unit), returnRoute, player, data, true).getFirst());
```

Comment:
Noting the APIs, it looks like we could improve it a bit and have more work be done by the 'callee' methods, noting:
```
getFuelCostsAndUnitsChargedFlatFuelCost(Collections.singleton(unit), returnRoute, player, data, true).getFirst()
```

In this example we could have the first arg take a `unit` directly, avoid the `true` parameter by creating a method overload, and could avoid the `getFirst` by having the method overload return that directly for us.

That would give something like:
```cost.add(getFuelCostsAndUnitsChargedFlatFuelCost(unit, returnRoute, player, data, true));
```

I think there is also a method like `computeIfAbsent` that could replace the if/else if structure for the map update. Going that route, if the body of the for loop were extracted to a helper method I think you could break this up into 2 or 3 pretty clean private methods.

Reply:
I actually originally created a method that I think was:
```
getFuelCostsIgnoringFlatFuel(Unit, Route, PlayerID, GameData)
```
I ended up not seeing much value in slightly cleaner parameters but adding misdirection given that the call is to a private method pretty much right below this one. If the call was to a public method then I'd completely agree. I'm open to adding an overloaded method back in if there is a strong consensus/opinion that it improves readability/maintainability.

Reply:
I will leave it up to you.

I like the cleaner and reduced parameters as it can make the method a bit more self-descriptive. If you can understand a method by it's name and args, then a person does not need to start hopping down the indirection tree. On the flip side, the boolean parameter being non-indirected requires reverse engineering of its usage to know what it really does.

Second comment/note I'll make, I tend to favor cleaner client code over server ("callee") code. If the client is not clean, then I'll iterate until it is. Then I drop down to the "callee" code and clean that up and it's dependent API's until that is also clean. This way down the whole stack there is no code you can't just read and understand.

Reply:
Third and final comment here:
Normally unit tests would be the driving factor in this decision. The *biggest* benefit of unit testing is for decoupling design, not for catching bugs.

Reply:
I'm inclined to leave as is for now since I'm not really strongly convinced one way is much better than the other. I agree with most of you approach and tend to do something fairly similar. And I think the public API calls of Route involving fuel are pretty decent. Unless you have any more specific questions/asks then I'd like to move forward with it and we can always revisit.